File: rb/features.rb
# FakeRest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FakeRestFeatures
  def self.make_feature(name)
    case name
    when "base"
      FakeRestBaseFeature.new
    when "ratelimit"
      FakeRestRatelimitFeature.new
    when "retry"
      FakeRestRetryFeature.new
    when "test"
      FakeRestTestFeature.new
    when "timeout"
      FakeRestTimeoutFeature.new
    else
      FakeRestBaseFeature.new
    end
  end
end

File: py/features.py
# FakeRest SDK feature factory

from feature.base_feature import FakeRestBaseFeature
from feature.test_feature import FakeRestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FakeRestBaseFeature(),
        "test": lambda: FakeRestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
